lib/cylc/config.py

```diff
@@ -725,6 +711,20 @@ def __init__(self, suite, fpath, template_vars=[], template_vars_file=None,
                                        '(graph the suite to see back-edges).')
         self.mem_log("config.py: end init config")
  
+    def is_graph_defined(self, dependency_map):
+        for item, value in dependency_map.items():
+            if item == 'graph':
+                # Async graph.
+                if value != '':
+                    return True
+            else:
+                # Cycling section.
+                for subitem, subvalue in value.items():
+                    if subitem == 'graph':
+                        if subvalue != '':
+                            return True
+        return False
+
```

Comment:
Made into a function to allow breakout from the nested loop via `return`.  The original only detected `&&` and `||` in async graphs - so I moved this to the graph string parsing function.